mesonbuild/backend/ninjabackend.py

```diff
@@ -2169,6 +2186,10 @@ def swift_module_file_name(self, target):
         return os.path.join(self.get_target_private_dir(target),
                             self.target_swift_modulename(target) + '.swiftmodule')
 
+    def swift_generated_header_file_name(self, target):
+        return os.path.join(self.get_target_private_dir(target),
+                            self.target_swift_modulename(target) + '-Swift.h')
```

Comment:
with `-working-directory` being set, this and the module path above seem to need an absolute path:

```suggestion
    def swift_generated_header_file_name(self, target):
        return os.path.join(self.get_target_private_dir_abs(target),
                            self.target_swift_modulename(target) + '-Swift.h')
```

Reply:
Does it? Works fine for me. These are used in the Ninja file where they're supposed to be relative to the top build dir, and in case of the header generation that isn't run in the private target dir but instead the top build dir.

Reply:
You're right of course, I had leftover local changes which affected this, so please ignore. Sorry for the noise.